test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -22,10 +22,11 @@ import scala.io.AnsiColor
 
```

Comment:
It would be nice to push more of the logic in this file into the `diff` package so the test reporter is really just handing over an actual and expected value and getting back what it should render.

Reply:
I would love to split `DefaultTestReporter` into to parts, where one translates test results into a set of simple (ish) domain model representing notions like assertion failures and diffs but without any markup stuff like lines, offsets and ANSI colors.
This would be useful for JUnitRunner and IDE integration @hmemcpy is working on, I think. 
And rendering with colors and offsets would be done by a different class.

But I think this should probably be a separate PR

Reply:
I did move `FailureRenderer`, `MessageMarkup` and `DiffRenderer` out into their own files.